drivers/dma/dmamux_stm32.c

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) 2019 Copyright (c) 2020 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Common part of DMAMUX drivers for stm32.
+ * @note  api functions named dmamux_stm32_
+ *        are calling the dma_stm32 corresponding function
+ *        implemented in dma_stm32.c
+ */
+
+#include <soc.h>
+#include <init.h>
+#include <drivers/dma.h>
+#include <drivers/clock_control.h>
+#include <drivers/clock_control/stm32_clock_control.h>
+
+#include "dmamux_stm32.h"
+#include "dma_stm32.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(dmamux_stm32, CONFIG_DMA_LOG_LEVEL);
+
+static bool stm32_dmamux_is_irq_happened(DMAMUX_Channel_TypeDef *dmamux,
+					u32_t id)
+{
+	if (func_ll_is_active_so[id](dmamux)) {
+		return true;
+	}
+	if (func_ll_is_active_rgo[id](dmamux)) {
+		return true;
+	}
+
+	return false;
+}
+
+static void dmamux_stm32_irq_handler(void *arg)
+{
+	struct device *dev = arg;
+	struct dmamux_stm32_data *mux_data = dev->driver_data;
+	const struct dmamux_stm32_config *mux_config =
+			dev->config->config_info;
+	DMAMUX_Channel_TypeDef *dmamux =
+			(DMAMUX_Channel_TypeDef *)(mux_config->base);
+	int i;
+
+	/* retrieve the dmamux channel */
+	for (i = 0; i < mux_config->channel_nb; i++) {
+		if (func_ll_is_active_so[i](dmamux)) {
+			break;
+		}
+		if (stm32_dmamux_is_irq_happened(dmamux, i)) {
```

Comment:
Maybe you can or this with the precedent

Reply:
yes
changed

Reply:
Apparently this code has been replaced by something else.
Can you possibly explain to me what happened?

Reply:
@erwango 
Can you confirm the code has changed here?

Reply:
@avisconti Yes, this was actually dead code, since dmamux irq is not used (it's only purpose for now is to be used as proxy), we don't need irq handling for now.